soak/prow/README.md

```diff
@@ -0,0 +1,116 @@
+6. Prowjob creates a Kubernetes Job resource in Soak test cluster and waits for the job to complete.
+7. Once the job completes, Prowjob cleans up the Job resource from #6 and the service controller resources from #4 .
+8. Based on the monitoring and alarming setup by service teams in the soak test cluster, they can review the results of
+   soak tests.
+
+## Onboarding Guide
+
+### Prerequisite
+* An AWS account to host the cluster. This will be reused from service teams onboarding for e2e test executions.
+* [yq](https://mikefarah.gitbook.io/yq/#install)
+* [eksctl](https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html)
+* [helm](https://helm.sh/docs/intro/install/)
+
+### Steps
+1. Update the current shell with AWS Credentials of ack-test role. Replace "<my-service-name>" with actual service name
+   in the command below.
+   ```bash
+   export SERVICE=<my-service-name> \
+   && export ACK_TEST_ROLE_NAME=$SERVICE-ack-test-role-DO-NOT-DELETE \
+   && export ACK_TEST_ROLE_ARN=$(aws iam get-role --role-name $ACK_TEST_ROLE_NAME | yq eval '.Role.Arn' -) \
+   && export AWS_CREDS_JSON=$(aws sts assume-role --role-arn $ACK_TEST_ROLE_ARN --role-session-name eks-cluster-create) \
+   && export AWS_ACCESS_KEY_ID=$(echo $AWS_CREDS_JSON | yq eval '.Credentials.AccessKeyId' -) \
+   && export AWS_SECRET_ACCESS_KEY=$(echo $AWS_CREDS_JSON | yq eval '.Credentials.SecretAccessKey' -) \
+   && export AWS_SESSION_TOKEN=$(echo $AWS_CREDS_JSON | yq eval '.Credentials.SessionToken' -) \
+   && export AWS_DEFAULT_REGION=us-west-2
+   ```
+
+2. Use `eksctl` to create an EKS cluster with assumed credentials
+   * Create a sample config file
+   ```bash
+      eksctl create cluster --name=soak-test-cluster --nodes=2 --version=1.20 --with-oidc --dry-run --enable-ssm > eksctl-config.yaml
+   ```
+   * Update the config as per your need
+   * Create the cluster. This may take ~10-20 minutes.
+   ```bash
+      eksctl create cluster --config-file=eksctl-config.yaml
+   ```
+
+3. Setup IRSA for soak test execution.
+   * Export AWS account id and OIDC provider for the EKS cluster
+   ```bash
+   export ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text) \
+   && export OIDC_PROVIDER=$(aws eks describe-cluster --name soak-test-cluster --query "cluster.identity.oidc.issuer" --output text | sed -e "s/^https:\/\///")
+   ```
+   * Create the trust relationship file for IRSA
+   ```bash
+   read -r -d '' TRUST_RELATIONSHIP <<EOF
+   {
+      "Version": "2012-10-17",
+      "Statement": [
+         {
+            "Effect": "Allow",
+            "Principal": {
+               "Federated": "arn:aws:iam::${ACCOUNT_ID}:oidc-provider/${OIDC_PROVIDER}"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+               "StringLike": {
+                  "${OIDC_PROVIDER}:sub": "system:serviceaccount:default:ack-*"
+               }
+            }
+         }
+      ]
+   }
+   EOF
+   ```
+   ```bash
```

Comment:
These can be in the same block?

Reply:
I want these command to be one click copy-paste from github repo and If I try to combine them together with "EOF" it just does not work properly.

So Keeping them as two separate one click copy block.